compiler/ballerina-backend-jvm/src/main/ballerina/compiler_backend_jvm/jvm_type_gen.bal

```diff
@@ -468,10 +470,43 @@ function createServiceType(jvm:MethodVisitor mv, bir:BObjectType objectType, bir
     mv.visitLdcInsn(typeDef.flags);
     mv.visitInsn(L2I);
 
-    // Initialize the object
-    mv.visitMethodInsn(INVOKESPECIAL, SERVICE_TYPE, "<init>",
-        io:sprintf("(L%s;L%s;I)V", STRING_VALUE, PACKAGE_TYPE),
-        false);
+    // initialize the object
+    if (strandIndex is int) {
```

Comment:
Shall we do this in two steps? - with second step only when required.
As in, extract this logic to a separate method. When you want set the annotation values, simply call the old method just like before, and then call the extracted method. So the initialization and setting annotation values happens in two steps.

Suggesting this to avoid having large methods that does multiple things..

Reply:
Kept the existing function as is and introduced another function which calls the existing function and then does annotation related stuff - https://github.com/ballerina-platform/ballerina-lang/commit/cb6db589f44e95f1f94e298fca2f0d0e80800e19#diff-eaee420729d9b10c600ac8582e8a3a99R470